clients/src/test/java/org/apache/kafka/clients/consumer/internals/CommitRequestManagerTest.java

```diff
@@ -382,9 +522,49 @@ private ClientResponse buildOffsetFetchClientResponse(
         return buildOffsetFetchClientResponse(request, topicPartitionData, error);
     }
 
+    private ClientResponse buildOffsetCommitClientResponse(final OffsetCommitResponse commitResponse,
+                                                           final Errors error) {
+        OffsetCommitResponseData data = new OffsetCommitResponseData();
+        OffsetCommitResponse response = new OffsetCommitResponse(data);
+        short apiVersion = 1;
+        return new ClientResponse(
+            new RequestHeader(ApiKeys.OFFSET_COMMIT, apiVersion, "", 1),
+            null,
+            "-1",
+            time.milliseconds(),
+            time.milliseconds(),
+            false,
+            null,
+            null,
+            commitResponse
+        );
+    }
+
+    public ClientResponse mockOffsetCommitResponse(String topic, int partition, short apiKeyVersion, Errors error) {
```

Comment:
Out of interest, why not use mockito here?

Reply:
😅 Could you be specific: Do you mean by doing something like this?

```
response = mock(ClientResponse.class);
when(response.receivedTimeMs()).thenReturn(...);
when(response.data()).thenReturn(responseData);
return response;
```